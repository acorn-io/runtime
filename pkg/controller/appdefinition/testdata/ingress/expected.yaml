kind: Namespace
apiVersion: v1
metadata:
  name: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    pod-security.kubernetes.io/enforce: baseline

---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "oneimage"
      "acorn.io/managed": "true"
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/container-name": "oneimage"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"alias":{},"build":{"context":".","dockerfile":"Dockerfile"},"image":"image-name","ports":[{"expose":true,"internalPort":81,"port":80,"protocol":"http"}],"sidecars":{"left":{"alias":{},"image":"foo","ports":[{"expose":true,"internalPort":91,"port":90,"protocol":"tcp"}]}}}'
    spec:
      terminationGracePeriodSeconds: 5
      enableServiceLinks: false
      shareProcessNamespace: true
      automountServiceAccountToken: false
      imagePullSecrets:
        - name: oneimage-pull-12345678
      containers:
        - name: oneimage
          image: "image-name"
          readinessProbe:
            tcpSocket:
              port: 81
          ports:
          - containerPort: 81
            protocol: "TCP"
        - name: left
          image: "foo"
          ports:
          - containerPort: 91
            protocol: "TCP"
          readinessProbe:
            tcpSocket:
              port: 91
---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: buildimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "buildimage"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "buildimage"
      "acorn.io/managed": "true"
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/container-name": "buildimage"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"alias":{},"build":{"context":".","dockerfile":"custom-dockerfile"},"image":"sha256:build-image","ports":[{"expose":true,"internalPort":81,"port":80,"protocol":"http"},{"expose":true,"internalPort":91,"port":443,"protocol":"tcp"}]}'
    spec:
      terminationGracePeriodSeconds: 5
      enableServiceLinks: false
      shareProcessNamespace: true
      automountServiceAccountToken: false
      imagePullSecrets:
        - name: buildimage-pull-12345678
      containers:
        - name: buildimage
          image: "sha256:build-image"
          readinessProbe:
            tcpSocket:
              port: 81
          ports:
            - containerPort: 81
              protocol: "TCP"
            - containerPort: 91
              protocol: "TCP"
---
kind: Service
apiVersion: v1
metadata:
  name: oneimage-publish-1234567890ab
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    "acorn.io/managed": "true"
spec:
  type: LoadBalancer
  ports:
    - port: 90
      targetPort: 91
      protocol: "TCP"
      name: "90"
  selector:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    acorn.io/managed: "true"
---
kind: Service
apiVersion: v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    "acorn.io/managed": "true"
spec:
  type: ClusterIP
  ports:
    - port: 80
      targetPort: 81
      protocol: "TCP"
      appProtocol: "HTTP"
      name: "80"
    - port: 90
      targetPort: 91
      protocol: "TCP"
      name: "90"
  selector:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    acorn.io/managed: "true"
---
kind: Service
apiVersion: v1
metadata:
  name: buildimage-publish-1234567890ab
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "buildimage"
    "acorn.io/managed": "true"
spec:
  type: LoadBalancer
  ports:
    - port: 443
      targetPort: 91
      protocol: "TCP"
      name: "443"
  selector:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "buildimage"
    acorn.io/managed: "true"
---
kind: Service
apiVersion: v1
metadata:
  name: buildimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "buildimage"
    "acorn.io/managed": "true"
spec:
  type: ClusterIP
  ports:
    - port: 80
      targetPort: 81
      protocol: "TCP"
      appProtocol: "HTTP"
      name: "80"
    - port: 443
      targetPort: 91
      protocol: "TCP"
      name: "443"
  selector:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "buildimage"
    acorn.io/managed: "true"
---
kind: Ingress
apiVersion: networking.k8s.io/v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    "acorn.io/managed": "true"
  annotations:
    acorn.io/container-name: oneimage
    acorn.io/hostnames: localhost,ci1.acorn.not
    acorn.io/port-number: "81"
spec:
  rules:
    - host: localhost
      http:
        paths:
          - backend:
              service:
                name: oneimage
                port:
                  number: 80
            path: /
            pathType: Prefix
    - host: ci1.acorn.not
      http:
        paths:
          - backend:
              service:
                name: oneimage
                port:
                  number: 80
            path: /
            pathType: Prefix
    - host: oneimage.app-name.app-namespace.local.on-acorn.io
      http:
        paths:
          - backend:
              service:
                name: oneimage
                port:
                  number: 80
            path: /
            pathType: Prefix
  tls:
  - hosts:
    - ci1.acorn.not
    secretName: test-tls-12345678
---
kind: Ingress
apiVersion: networking.k8s.io/v1
metadata:
  name: buildimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "buildimage"
    "acorn.io/managed": "true"
  annotations:
    acorn.io/container-name: buildimage
    acorn.io/hostnames: buildimage.app-name.app-namespace.local.on-acorn.io
    acorn.io/port-number: "81"
spec:
  rules:
    - host: buildimage.app-name.app-namespace.local.on-acorn.io
      http:
        paths:
          - backend:
              service:
                name: buildimage
                port:
                  number: 80
            path: /
            pathType: Prefix
---
kind: AppInstance
apiVersion: acorn.io/v1
metadata:
  uid: "1234567890abc"
  name: app-name
  namespace: app-namespace
spec:
  publishProtocols:
    - tcp
    - http
  image: test
  endpoints:
    - target: oneimage
      hostname: localhost
    - target: oneimage
      hostname: ci1.acorn.not
status:
  namespace: app-created-namespace
  appSpec:
    containers:
      oneimage:
        sidecars:
          left:
            image: "foo"
            ports:
              - port: 90
                internalPort: 91
                protocol: tcp
                expose: true
        ports:
          - port: 80
            internalPort: 81
            expose: true
            protocol: http
        image: "image-name"
        build:
          dockerfile: "Dockerfile"
          context: "."
      buildimage:
        ports:
          - port: 80
            internalPort: 81
            expose: true
            protocol: http
          - port: 443
            internalPort: 91
            expose: true
            protocol: tcp
        image: "sha256:build-image"
        build:
          dockerfile: "custom-dockerfile"
          context: "."
  conditions:
    defined:
      success: true
---
apiVersion: v1
data:
  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUR6ekNDQXJlZ0F3SUJBZ0lVZWtRUWRTNXluTTUwUVVDZjBMOHVtd1lOUUJRd0RRWUpLb1pJaHZjTkFRRUwKQlFBd1R6RUxNQWtHQTFVRUJoTUNWVk14RGpBTUJnTlZCQWdUQlZSbGJYQmxNUkF3RGdZRFZRUUhFd2RCY21sNgpiMjVoTVE0d0RBWURWUVFLRXdWQlkyOXliakVPTUF3R0ExVUVBeE1GUTBrZ1EwRXdIaGNOTWpJd05USTBNakUwCk5UQXdXaGNOTWpNd05USTBNakUwTlRBd1dqQlNNUXN3Q1FZRFZRUUdFd0pWVXpFT01Bd0dBMVVFQ0JNRlZHVnQKY0dVeEVEQU9CZ05WQkFjVEIwRnlhWHB2Ym1FeERqQU1CZ05WQkFvVEJVRmpiM0p1TVJFd0R3WURWUVFERXdoQgpZMjl5YmlCRFNUQ0NBU0l3RFFZSktvWklodmNOQVFFQkJRQURnZ0VQQURDQ0FRb0NnZ0VCQUtaOVNSMURyQmtPCm1LeUVnT1l4a0pnRnFGNWlSS05JWUJmN3hrTHI3cWN6aHQ1YkVxZFRUa3pNL1RlMHc3RGZzeUlYMWFIZ21NdE0KUk53TnBvY013ZmpRTElHbmJWNE1waVNQV204cnRQQTl6NURhNEdPeVkyUGFkSkV1d1NsRUVuV2d2enN1eGJ2SQpLMUQ1aWdKdUp0eHJuZUo3VktablpMTzRQWGhoL0dIVmEvODhwVnM5NlJ0YzIzSDh6emdNK2Fjazk0V0lBUi9XClBVbThTYW50eDJzR2JrRTNxNTYwejBLdmxsT2x0WXVUZUIydUduZUEycm5KWExQK0c4M2s2Y3NLMW9Yb2t4UmIKTTdLMmpwZHRlcmdOczFRRG1HTGVHbHJLR0swZnRmSVk3d2trckg3NEx4c29aSzB3T3VsM3psKzRJYnR4d0JMYwpvZWdXb0pSYzBrY0NBd0VBQWFPQm56Q0JuREFPQmdOVkhROEJBZjhFQkFNQ0JhQXdFd1lEVlIwbEJBd3dDZ1lJCkt3WUJCUVVIQXdFd0RBWURWUjBUQVFIL0JBSXdBREFkQmdOVkhRNEVGZ1FVbWZ1aFpjbEpkcU5PTHBSTHVPVVgKYUZQbHpFTXdId1lEVlIwakJCZ3dGb0FVamEzWTlnQU5vZGJ5WnhyZllpaUZOYjRqTkZNd0p3WURWUjBSQkNBdwpIb0lOWTJreExtRmpiM0p1TG01dmRJSU5ZMmt5TG1GamIzSnVMbTV2ZERBTkJna3Foa2lHOXcwQkFRc0ZBQU9DCkFRRUF0Wno2SEhMc1kzdjlKK0dQbEpickFuRDNFUURwMmdiaklSSmVFcEVlNnd4ZGdjOUxtc0djR3dCSVBpaHQKN2tEanBjNmJNdG5HTmFNUlBabWkrZXhZU3BGZVQyMVFrVVZsUlhwKzlpcG94OW8rNDFJb1B4Mm9xeHBRRGpLbwpxaDJpQ2FUT21PZkpkRGdzbkZZTTh5V1hkU1R4Z0pEandrblRaRmlSVG84c1JPc3FDSklOVGVIclhHaVRiVTUyCkJGUDMyQWp4bk82U1FZYSswNnJlT0hqaUdkc1dwUzllMTJmRzh4T0IzNDY4b2gwRC9NTjhITUE1SlZlWEE4K04KTklqT0NNb1FUZjFYbzRTVnV6bGQxNzlMV0p2SjJSNWdkZGFHZjZMNkovSXE2NlFtTFljckhMUkRRSEdBVWdIVApHbStOZWFaK3EyREJjSGVaQlVoK3FmVTR1UT09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
  tls.key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcEFJQkFBS0NBUUVBcG4xSkhVT3NHUTZZcklTQTVqR1FtQVdvWG1KRW8waGdGL3ZHUXV2dXB6T0czbHNTCnAxTk9UTXo5TjdURHNOK3pJaGZWb2VDWXkweEUzQTJtaHd6QitOQXNnYWR0WGd5bUpJOWFieXUwOEQzUGtOcmcKWTdKalk5cDBrUzdCS1VRU2RhQy9PeTdGdThnclVQbUtBbTRtM0d1ZDRudFVwbWRrczdnOWVHSDhZZFZyL3p5bApXejNwRzF6YmNmelBPQXo1cHlUM2hZZ0JIOVk5U2J4SnFlM0hhd1p1UVRlcm5yVFBRcStXVTZXMWk1TjRIYTRhCmQ0RGF1Y2xjcy80YnplVHB5d3JXaGVpVEZGc3pzcmFPbDIxNnVBMnpWQU9ZWXQ0YVdzb1lyUisxOGhqdkNTU3MKZnZndkd5aGtyVEE2NlhmT1g3Z2h1M0hBRXR5aDZCYWdsRnpTUndJREFRQUJBb0lCQUhwaGNkTXZJVnd4M0l4RAp3alJ6anFRUTFKaThoMU8zS2R1M1dSOXNLanVCcHN2Z0NqQXhEa2RxRDJCWllkZXNPQk1CamltMTNOcmpyMkpLCkxuVHJMZEpsdzdGeWp2UXBQQjkzdUxRclhvK1IvL2VzekhDOGVrNjJFQ1dBUnNDOHB3ME1Ja1d3dGV5NDBRNXUKdjhIVmtuOUlKQUNOUThZbURtSUZOam85WExYVWFoaEhHQVpvSnBsVzR2WkpRelhIMWRjK3dLSGY0cjhjZ0laRgppMmh5TkdzRUEyRFBOemNCNHQzaitEZXNrUzNIOW1PSTVld1V2c1BUdStFWVdMbjhEOTRKOUlNNlZ5QW8vcFRaCjhNS2orTUxTUkFnNGJjMUVZbUM0VVg2dVl2VVpZSmkvY3QyMWtrRFc3ZEtlZU9OdjEwVkV2VkIrMUlES3krMGkKN0t5alZ1RUNnWUVBMW9OMUIxeTUvWEZnV1hZTHZGb3ovY0NKdFJjSnNxUWRxRXRoZG9BYUdVRTJBU3BCL3lHNgpQR3M1UXdsRjhTUmdERXRKRzlOMEU4TXBtaXplTWc0MFI3cDJoZ2lXN0xQWWM4VE84Y0I1SDdNMzFOTGNVL0VaCm9qTEVPOUVYdlJBTThVVmJ0TUQzdXpJbTFJckRBdGI1dnJ6MmF3SFd6VG9TVm1WZDFxR3A4RDhDZ1lFQXhyQW8KMTM2K0xiM3F0dkxWQ2hKUzVBanVwZ1pLMTRHdithTlFVM0h3M0ZDYXhlL0NpcXNUN0NFOUFkbFc4N0pOeWVlaQpJWEZab3FwTS9VN0RMdnFPRjZkcU0rb3daempRWW1SU0dSbk5OeHpocHJiK08yUU9ZUnZSYnRlaHVJV1BHZ09DCjJHZC9QRzNhQUdia1BFMSsrS21YTFVaOXVmKzVqVVJ0QUNka20va0NnWUVBcjhDdTNxUlRZbHBFNWFjNUFFNHIKY01rU1NvT2dsM090Tk5qbDlwQUlVZzdDcjN1dTdOajJYRkJCYnpJbTB1YkpwdWo3OGtkeVFFclRTQlZneTBRbQpBMnE3ZHgrTWdFenRtQlIwdFUvTUxYUEoxNTRYbE5MaC9LbDVhOFRwblNhTVpCVStpTDdkZWNzaVUxdUpZVWMvCkFjNlNXcVA4R05OdWVNaStkSDZwM0VNQ2dZQmFnNllYaGdYZGtNZDgwT044MHh3d0JtR3lBT3ZYZ1oxSmZMejcKUnMwVGtTVHhSMmk1QTNidis3UnZRelA1a1NFZStnZENkUUVBdWRTWkVXaVoza3NkTzRlR1h3bnQvRlB5ZndGbgpyZmx2UWhrS2VoTVh1MTFSbi9xcEpGTG9YTy8wU2VVeDhQUmw4eVY5U1dKd2xsMWhxdURVSDJqbmN6aDB3dGRsCldvemJBUUtCZ1FDMWJEaTVJV1dDODhBcmNVa0hZcmVKdHQyWGQ5dGczY2RKcjJTMkNCT2RKNkt1T2xvMml4c0oKWDBtZ25HbGpLb1JrTjhScWc1anlBK29tOVN6TWlmZzF4dGQ4cHp1VlJ2QVFnMXRua3RJNTFoV1pQOXpvdklESwpBK3lTQVE0dG1zb0lrRW9URXE0Zkxva0ZvSGRpcTl6bi9TamJRamNhUmxiZlFUVFFDQzhNd2c9PQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
kind: Secret
metadata:
  name: test-tls-12345678
  namespace: app-created-namespace
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-uid: 1234567890abc
    acorn.io/managed: "true"
    acorn.io/secret-generated: "true"
    acorn.io/secret-name: test-tls
type: kubernetes.io/tls
---
kind: Secret
apiVersion: v1
metadata:
  name: buildimage-pull-12345678
  namespace: app-created-namespace
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
type: "kubernetes.io/dockerconfigjson"
data:
  ".dockerconfigjson": eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
---
kind: Secret
apiVersion: v1
metadata:
  name: oneimage-pull-12345678
  namespace: app-created-namespace
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
type: "kubernetes.io/dockerconfigjson"
data:
  ".dockerconfigjson": eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
---
