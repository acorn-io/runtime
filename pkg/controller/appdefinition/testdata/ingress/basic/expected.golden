`apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: buildimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: buildimage
  namespace: app-created-namespace

---
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: buildimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: buildimage
  namespace: app-created-namespace
spec:
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: buildimage
      acorn.io/managed: "true"
  strategy: {}
  template:
    metadata:
      annotations:
        acorn.io/container-spec: '{"build":{"context":".","dockerfile":"custom-dockerfile"},"image":"sha256:build-image","metrics":{},"ports":[{"port":80,"protocol":"http","publish":true,"targetPort":81},{"port":443,"protocol":"tcp","publish":true,"targetPort":91}],"probes":null}'
        karpenter.sh/do-not-evict: "true"
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/container-name: buildimage
        acorn.io/managed: "true"
        acorn.io/project-name: app-namespace
    spec:
      containers:
      - image: sha256:build-image
        name: buildimage
        ports:
        - containerPort: 81
          protocol: TCP
        - containerPort: 91
          protocol: TCP
        readinessProbe:
          tcpSocket:
            port: 81
        resources: {}
      enableServiceLinks: false
      hostname: buildimage
      imagePullSecrets:
      - name: buildimage-pull-1234567890ab
      serviceAccountName: buildimage
      terminationGracePeriodSeconds: 5
status: {}

---
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: buildimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: buildimage
  namespace: app-created-namespace
spec:
  maxUnavailable: 25%
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: buildimage
      acorn.io/managed: "true"
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0

---
apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: oneimage
  namespace: app-created-namespace

---
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: oneimage
  namespace: app-created-namespace
spec:
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: oneimage
      acorn.io/managed: "true"
  strategy: {}
  template:
    metadata:
      annotations:
        acorn.io/container-spec: '{"build":{"context":".","dockerfile":"Dockerfile"},"image":"image-name","metrics":{},"ports":[{"port":80,"protocol":"http","publish":true,"targetPort":81}],"probes":null,"sidecars":{"left":{"image":"foo","metrics":{},"ports":[{"port":90,"protocol":"tcp","publish":true,"targetPort":91}],"probes":null}}}'
        karpenter.sh/do-not-evict: "true"
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/container-name: oneimage
        acorn.io/managed: "true"
        acorn.io/project-name: app-namespace
    spec:
      containers:
      - image: image-name
        name: oneimage
        ports:
        - containerPort: 81
          protocol: TCP
        readinessProbe:
          tcpSocket:
            port: 81
        resources: {}
      - image: foo
        name: left
        ports:
        - containerPort: 91
          protocol: TCP
        readinessProbe:
          tcpSocket:
            port: 91
        resources: {}
      enableServiceLinks: false
      hostname: oneimage
      imagePullSecrets:
      - name: oneimage-pull-1234567890ab
      serviceAccountName: oneimage
      terminationGracePeriodSeconds: 5
status: {}

---
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: oneimage
  namespace: app-created-namespace
spec:
  maxUnavailable: 25%
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: oneimage
      acorn.io/managed: "true"
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0

---
apiVersion: internal.acorn.io/v1
kind: ServiceInstance
metadata:
  annotations:
    acorn.io/app-generation: "0"
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: buildimage
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.buildimage
  name: buildimage
  namespace: app-created-namespace
spec:
  appName: app-name
  appNamespace: app-namespace
  container: buildimage
  default: false
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: buildimage
    acorn.io/managed: "true"
  ports:
  - port: 80
    protocol: http
    publish: true
    targetPort: 81
  - port: 443
    protocol: tcp
    publish: true
    targetPort: 91
status: {}

---
apiVersion: internal.acorn.io/v1
kind: ServiceInstance
metadata:
  annotations:
    acorn.io/app-generation: "0"
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.oneimage
  name: oneimage
  namespace: app-created-namespace
spec:
  appName: app-name
  appNamespace: app-namespace
  container: oneimage
  default: false
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
  ports:
  - port: 80
    protocol: http
    publish: true
    targetPort: 81
  - port: 90
    protocol: tcp
    publish: true
    targetPort: 91
  publish:
  - hostname: localhost
  - hostname: ci1.acorn.not
status: {}

---
apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: buildimage-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: oneimage-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: internal.acorn.io/v1
kind: AppInstance
metadata:
  creationTimestamp: null
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
  ports:
  - hostname: localhost
    publish: true
    serviceName: localhost
    targetServiceName: oneimage
  - hostname: ci1.acorn.not
    publish: true
    serviceName: ci1.acorn.not
    targetServiceName: oneimage
status:
  appImage:
    buildContext: {}
    id: test
    imageData: {}
    vcs: {}
  appSpec:
    containers:
      buildimage:
        build:
          context: .
          dockerfile: custom-dockerfile
        image: sha256:build-image
        metrics: {}
        ports:
        - port: 80
          protocol: http
          publish: true
          targetPort: 81
        - port: 443
          protocol: tcp
          publish: true
          targetPort: 91
        probes: null
      oneimage:
        build:
          context: .
          dockerfile: Dockerfile
        image: image-name
        metrics: {}
        ports:
        - port: 80
          protocol: http
          publish: true
          targetPort: 81
        probes: null
        sidecars:
          left:
            image: foo
            metrics: {}
            ports:
            - port: 90
              protocol: tcp
              publish: true
              targetPort: 91
            probes: null
  appStatus: {}
  columns: {}
  conditions:
    reason: Success
    status: "True"
    success: true
    type: defined
  defaults: {}
  namespace: app-created-namespace
  staged:
    appImage:
      buildContext: {}
      imageData: {}
      vcs: {}
  summary: {}
`
